com.avaloq.tools.ddk.xtext/src/com/avaloq/tools/ddk/xtext/scoping/InferredImplicitReferencesAdapter.java

```diff
@@ -62,26 +84,41 @@ public void addImplicitReference(final URI uri) {
   }
 
   /**
-   * Returns all implicit references registered with this adapter.
+   * Adds an inferred reference to the given URI (object or resource URI).
+   *
+   * @param resourceUri
+   *          referenced resource
+   */
+  public void addInferredReference(final URI resourceUri) {
+    inferredResources.add(resourceUri);

```

Comment:
the name is a bit misleading, imho. The actual resources aren't really inferred. Perhaps inferenceDependencies would fit a bit more? The name of the method can also be adjusted accordingly, addInferenceDependency or smth like that